test/e2e/burnFees.test.ts

```diff
@@ -42,8 +42,8 @@ describe("E2E: Burn fees", () => {
     const { chainId } = await ethers.provider.getNetwork();
     domainSeparator = domain(chainId, settlement.address);
 
-    owl = await waffle.deployContract(deployer, ERC20, ["OWL", 18]);
-    dai = await waffle.deployContract(deployer, ERC20, ["DAI", 18]);
+    owl = await waffle.deployContract(deployer, ERC20, ["OWL", "Owl token"]);
+    dai = await waffle.deployContract(deployer, ERC20, ["DAI", "Dai token"]);
```

Comment:
changing the decimals to a token name? Weird...

Reply:
I was looking at old code to see how to deploy an ERC token and found this code. Then I saw my tests failing and by debugging I realized that the second input field is the name of the token and not the number of decimals. To avoid this in the future I changed all confusing occurrences like this.
This is also why I added `src/contracts/test/TestERC20.sol`, apparently we didn't e2e test using tokens with custom decimals before.

Reply:
:see_no_evil: 

> apparently we didn't e2e test using tokens with custom decimals before.

We do, there are some USDC tests where it assumes 6 decimals. I think the reason we never caught this is that the `decimals` value is purely for UI purposes an does not affect any computation in any way. So if, for an E2E test I set up and order trading `1e18 ETH` for `3000e6 USDC`, then even if `await usdc.decimals()` is `18`, it is, for all other intents and purposes, acting like a 6 decimal token because of the arbitrary value assigned to it.